File: test/end-to-end/Point/Point.ts
//
export class Point {
    public x: number;
    public y: number;
    protected square: number;
    private name: string;

    public constructor(x: number, y: number) {
        this.x = x;
        this.y = y;
        this.square = x * y;
        this.name = "";
    }
}
//

File: test/end-to-end/Point/Point.rb
#
class Point
    attr_accessor: x
    attr_accessor: y
    attr_accessor: square
    attr_accessor: name

    def initialize(x, y)
        self.x = x
        self.y = y
        self.square = x * y
        self.name = ""
    end
end
#
